challenges/Python_Parallel_HDF5/README.md

```diff
@@ -0,0 +1,230 @@
+#
+# To activate this environment, use
+#
+#     $ conda activate /ccsopen/proj/<YOUR_PROJECT_ID>/<YOUR_USER_ID>/conda_envs/ascent/h5pympi-ascent
+#
+# To deactivate an active environment, use
+#
+#     $ conda deactivate
+```
+
+Due to the specific nature of conda on Ascent, we will be using `source activate` instead of `conda activate` to activate our new environment:
+
+```
+$ source activate /ccsopen/proj/<YOUR_PROJECT_ID>/<YOUR_USER_ID>/conda_envs/ascent/h5pympi-ascent
+```
+
+The path to the environment should now be displayed in "( )" at the beginning of your terminal lines, which indicate that you are currently using that specific conda environment. 
+If you check with `conda env list`, you should see that the `*` marker is next to your new environment, which means that it is currently active:
+
+```
+$ conda env list
+
+# conda environments:
+#
+                      *  /ccsopen/proj/<YOUR_PROJECT_ID>/<YOUR_USER_ID>/conda_envs/ascent/h5pympi-ascent
+base                     /sw/ascent/python/3.6/anaconda3/5.3.0
+```
+
+## Installing mpi4py
+
+Now that we have a fresh conda environment, we will next install mpi4py from source into our new environment.
+To make sure that we are building from source, and not a pre-compiled binary, we will be using pip:
+
+```
+$ MPICC="mpicc -shared" pip install --no-binary=mpi4py mpi4py
+```
+
+The `MPICC` flag ensures that you are using the correct C wrapper for MPI on the system.
+Building from source typically takes longer than a simple `conda install`, so the download and installation may take a couple minutes.
+If everything goes well, you should see a "Successfully installed mpi4py" message.
+
+## Installing h5py
+
+Next, we will install h5py from source.
+Because h5py depends on NumPy, we will install an optimized version of the NumPy package first using `conda install`:
+
+```
+$ conda install -c defaults --override-channels numpy
+```
+
+The `-c defaults --override-channels` flags ensure that conda will search for NumPy only on the "defaults" channel.
+Installing NumPy in this manner results in an optimized NumPy that is built against linear algebra libraries, which performs operations much faster.
+
+Next, we are finally ready to install h5py from source:
+
+```
+$ HDF5_MPI="ON" CC=gcc pip install --no-binary=h5py h5py
+```
+
+The `HDF5_MPI` flag is the key to telling pip to build h5py with parallel support, while the `CC` flag makes sure that we are using the correct C wrapper.
+This installation will take much longer than both the mpi4py and NumPy installations (5+ minutes if the system is slow).
+When the installation finishes, you will see a "Successfully installed h5py" message.
+
+## Testing parallel h5py
+
+Now for the fun part, testing to see if our build was truly successful.
+We will test our build by trying to write an HDF5 file in parallel using 42 MPI tasks.
+First, let's launch an interactive batch job:
+
+```
+$ bsub -W 0:15 -nnodes 1 -P <YOUR_PROJECT_ID> -Is $SHELL
```

Comment:
I don't think we should have this in an interactive session since there are only 16 total nodes and the number of participants can be large. Don't want to create long wait times in the queues :)